compiler.go

```diff
@@ -87,10 +93,18 @@ type compiler struct {
 	codeScratchpad []instruction
 }
 
+func (c *compiler) getScriptOrModule() interface{} {
+	if c.module != nil {
+		return c.module
+	}
+	return c.p // TODO figure comething better
+}
+
 type binding struct {
 	scope        *scope
 	name         unistring.String
 	accessPoints map[*scope]*[]int
+	getIndirect  func(vm *vm) Value
```

Comment:
This used to look really hacky when I first added it, but over time and fixing all kinds of issues I don't see any way to:
1. have it work with all the different corner cases with functions being able to be exported, the differences between var and let/const and so on. 
2. Have a compiled set of modules that you can evaluate on different Runtimes with no problems
3. Not have to rewrite huge parts of goja - and I don't even know what I will have to do to make it work, but I guess something that at the beginging of execution(?) changes to not use `getDirect` but something more concrete.

I have also tried to only make the smallest change needed in all the places and it seems to work for everything tried ... but there might be bugs that are just not covered by tests.